trunk/source/Library-Proofs/src/de/uni_freiburg/informatik/ultimate/lib/proofs/floydhoare/FloydHoareUtils.java

```diff
@@ -127,6 +127,29 @@ public static void createInvariantResults(final String pluginName, final IIcfg<I
 			}
 			new WitnessInvariant(invResult.getInvariant()).annotate(locNode);
 		}
+		
+		// Temporary hack: Get the invariant of the pre-error location
+		// There exists only one in our use case, needs to be generalized if obtaining
+		// invariants for all pre-error locations is desired
+		final var errorLocs = IcfgUtils.getErrorLocations(icfg).iterator();
+		if (errorLocs.hasNext())
+		{
+			final var preLocs = errorLocs.next().getIncomingNodes();
+			// Hack: For reqcheck there only exists one pre-error location
+			final var preLoc = preLocs.isEmpty() ? null : preLocs.get(0);
+			if (preLoc != null)
+			{
+				final IPredicate hoare = annotation.getAnnotation(preLoc);
+				if (hoare == null)
+				{
+					return;
+				}
+				final Term formula = hoare.getFormula();
+				final var invResult =
+						new InvariantResult<IIcfgElement>(pluginName, preLoc, backTranslatorService, formula, checks);
+				reporter.accept(invResult);
+			}
+		}
```

Comment:
@maul-esel : I guess the "correct" way would just be the disjunction of all location invariants of the predecessors, right? I would think that we can assume that all variables are in still in scope.

Reply:
The correct way to do what?

At the moment, the code creates an invariant result for the predecessor location. I don't think we should build a disjunction for this result. Since I don't fully understand the application, I can't really say how these potentially multiple invariants should be combined (but that would happen somewhere in `Pea2Boogie`, I guess).

Reply:
To support more than one predecessor :) 
But you are right, one could also generate multiple results, one for each location. I just dont like the idea to have a half-finished implementation here.

Reply:
Closing and moving this to https://github.com/ultimate-pa/ultimate/pull/681#discussion_r1783314091